velox/core/Expressions.cpp

```diff
@@ -364,7 +364,9 @@ bool equalsImpl(
 
   if (otherNull || thisNull) {
     return BaseVector::compareNulls(thisNull, otherNull, kEqualValueAtFlags)
-        .value();
+                .value() == 0
+        ? true
+        : false;
```

Comment:
nit: Just `return BaseVector::compareNulls(thisNull, otherNull, kEqualValueAtFlags).value() == 0;` should be enough.